com.unity.render-pipelines.high-definition/Editor/VFXGraph/VFXHDRPSubTarget.cs

```diff
@@ -285,8 +285,16 @@ enum VFXAttributeType
             }
             defineSpaceDescriptor = new AdditionalCommandDescriptor("VFXDefineSpace", defineSpaceDescriptorContent);
 
+            //Texture used as input of the shaderGraph will be declared by the shaderGraph generation
+            //However, if we are sampling a texture (or a mesh), we have to declare them before the VFX code generation.
+            //Thus, remove texture used in SG from VFX declaration and let the remainder.
+            var filteredTextureInSG = context.inputSlots.Where(o =>
+            {
+                return VFXExpression.IsTexture(o.property.type);
+            }).Select(o => o.property.name);
```

Comment:
N.B.: Check double texture sampling (in SG & VFX)

Reply:
Checked locally, it works as expected @julienf-unity (I suppose we are duplicating the mapping in that case, the same texture is bound twice in the material) :
![_read_texture_from_same_texture](https://user-images.githubusercontent.com/22494706/131513772-a0e52730-4711-426a-a96a-817541219b14.gif)